test/unit/Quoter.t.sol

```diff
@@ -0,0 +1,159 @@
+pragma solidity ^0.8.0;
+
+import "v3-core/test/__fixtures/BaseTest.sol";
+
+import { WETH } from "solmate/tokens/WETH.sol";
+import { Math } from "@openzeppelin/utils/math/Math.sol";
+import { FixedPointMathLib } from "solmate/utils/FixedPointMathLib.sol";
+
+import { MathUtils } from "v3-core/src/libraries/MathUtils.sol";
+
+import { Quoter, ExtraData } from "src/Quoter.sol";
+import { ReservoirLibrary } from "src/libraries/ReservoirLibrary.sol";
+
+contract QuoterTest is BaseTest
+{
+    WETH    private _weth   = new WETH();
+    Quoter  private _quoter = new Quoter(address(_factory), address(_weth));
+
+    function testQuoteAddLiquidity(uint256 aAmountAToAdd, uint256 aAmountBToAdd) public
+    {
+        // assume
+        uint256 lAmountAToAdd = bound(aAmountAToAdd, 1000, type(uint112).max);
+        uint256 lAmountBToAdd = bound(aAmountBToAdd, 1000, type(uint112).max);
+
+        // act
+        (uint256 lAmountAOptimal, uint256 lAmountBOptimal, uint256 lLiq)
+            = _quoter.quoteAddLiquidity(address(_tokenA), address(_tokenB), 0, lAmountAToAdd, lAmountBToAdd);
+
+        // assert
+        assertEq(lAmountAOptimal, Math.min(lAmountAToAdd, lAmountBToAdd));
+        assertEq(lAmountBOptimal, lAmountAOptimal);
+        assertEq(lLiq, FixedPointMathLib.sqrt(lAmountAOptimal * lAmountBOptimal));
+    }
+
+    function testQuoteAddLiquidity_Stable(uint256 aAmountAToAdd, uint256 aAmountBToAdd) public
+    {
+        // assume
+        uint256 lAmountAToAdd = bound(aAmountAToAdd, 1000, type(uint112).max);
+        uint256 lAmountBToAdd = bound(aAmountBToAdd, 1000, type(uint112).max);
+
+        // act
+        (uint256 lAmountAOptimal, uint256 lAmountBOptimal, uint256 lLiq)
+            = _quoter.quoteAddLiquidity(address(_tokenA), address(_tokenB), 1, lAmountAToAdd, lAmountBToAdd);
+
+        // assert
+        assertEq(lAmountAOptimal, Math.min(lAmountAToAdd, lAmountBToAdd));
+        assertEq(lAmountBOptimal, lAmountAOptimal);
+        assertEq(lLiq, lAmountAOptimal + lAmountBOptimal);
```

Comment:
I've been thinking, is `liq = a + b` what balancer and others do? It feels like it will have liquidity spike massively when a pool is imbalanced? I guess this is countered by the fact its a TWAL?